perfkitbenchmarker/flag_util.py

```diff
@@ -304,3 +306,60 @@ def DEFINE_units(name, default, help, convertible_to=None,
   serializer = UnitsSerializer()
 
   flags.DEFINE(parser, name, default, help, flag_values, serializer, **kwargs)
+
+
+# The YAML flag type is necessary because flags can be read either via
+# the command line or from a config file. If they come from a config
+# file, they will already be parsed as YAML, but if they come from the
+# command line, they will be raw strings. The point of this flag is to
+# guarantee a consistent representation to the rest of the program.
+class YAMLParser(flags.ArgumentParser):
+  """Parse a flag containing YAML."""
+
+  syntactic_help = 'A YAML expression. Ex: "{x: y, foo: 2.3}".'
+
+  def Parse(self, inp):
+    """Parse the input.
+
+    Args:
+      inp. A string or the result of yaml.load. If a string, should be
+      a valid YAML document.
+    """
+
+    if isinstance(inp, str):
+      # This will work unless the user writes a config with a quoted
+      # string that, if unquoted, would be parsed as a non-string
+      # Python type (example: '123'). In that case, the first
+      # yaml.load() in the config system will strip away the quotation
+      # marks, and this second yaml.load() will parse it as the
+      # non-string type. However, I think this is the best we can do
+      # without significant changes to the config system, and the
+      # problem is unlikely to occur in PKB.
+      try:
+        return yaml.load(cStringIO.StringIO(inp))
```

Comment:
Because `yaml.load()` needs a stream, not a string. I'll add a comment about that.